extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/exec/MSQSelectTest.java

```diff
@@ -1086,6 +1091,65 @@ public void testGroupByMultiValueMeasureQuery()
         .verifyResults();
   }
 
+  @Test
+  public void testGroupByOnFooWithDurableStoragePathAssertions()
+  {
+    RowSignature rowSignature = RowSignature.builder()
+                                            .add("cnt", ColumnType.LONG)
+                                            .add("cnt1", ColumnType.LONG)
+                                            .build();
+
+    ExecutorService executorService = Execs.singleThreaded("path-verifier");
+    final AtomicBoolean existsOnce = new AtomicBoolean(false);
+    executorService.submit(() -> {
+      while (true) {
+        File successFile = new File(
```

Comment:
This logic is very racy. Can we intercept calls on the local file storage connector and then assert on the paths ?